### PR TITLE
Jordan/allow for async loading of vue components

### PR DESCRIPTION
**Pull request by jbibla**
Closes #ISSUE

**Description:**

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by faboweb** [comment]
Interesting! Does this bring a performance improvement?

**Comment by jbibla** [comment]
> Does this bring a performance improvement?

yes! that's the idea! instead of downloading the whole bundle on first visit the user only downloads the js files for the components that are needed (code-splitting). webpack will only do this if we load components this way.

**Comment by faboweb** [comment]
I meant, did you measure the performance improvement?

**Comment by jbibla** [comment]
> did you measure the performance improvement?

yes, i'm trying to find the optimal setup.

one major issue is how large cosmos-ledger and cosmos-keys are. trying to figure out why these are loading on initialization instead of as needed. but even so, can we reduce the size of these libraries?

File: src/ActionModal/components/ActionModal.vue [hunk 1]
```diff
@@ -301,7 +301,7 @@ import { track } from "scripts/google-analytics"
 import { UserTransactionAdded } from "src/gql"
 import config from "src/../config"
 
-const ActionManager = () => import("../utils/ActionManager")
+const ActionManager = () => import(/* webpackChunkName: "ActionManager" */ "../utils/ActionManager")
```
Comment (faboweb) [inline src/ActionModal/components/ActionModal.vue]: that looks dirty ^^' is this necessary?

**Comment by faboweb** [comment]
> one major issue is how large cosmos-ledger and cosmos-keys are. trying to figure out why these are loading on initialization instead of as needed. but even so, can we reduce the size of these libraries?

probably we can, but I don't consider this anyway near of our prios. what do you think?

File: src/components/staking/PageValidators.vue [hunk 2]
```diff
@@ -40,11 +40,11 @@
...
-import TmDataLoading from "common/TmDataLoading"
-import TableValidators from "staking/TableValidators"
-import TmPage from "common/TmPage"
-import TmField from "common/TmField"
-import TmBtn from "common/TmBtn"
+const TmDataLoading = () => import("common/TmDataLoading")
+const TableValidators = () => import("staking/TableValidators")
+const TmPage = () => import("common/TmPage")
+const TmField = () => import("common/TmField")
+const TmBtn = () => import("common/TmBtn")
```
Comment (colw) [inline src/components/staking/PageValidators.vue]: This is _probably_ overkill. But curious once we test it anyway.

File: src/ActionModal/components/ActionModal.vue (hunk 1 above)
Comment (jbibla) [reply src/ActionModal/components/ActionModal.vue]: no just using chunk names for testing and debugging

File: src/routes.js [hunk 3]
```diff
@@ -17,7 +17,7 @@ export default [
     meta: {
       feature: "Proposals"
     },
-    component: require(`./components/governance/PageProposals`).default
+    component: () => import(`./components/governance/PageProposals`)
```
Comment (faboweb) [inline src/routes.js]: This is already loading the pages and there content async

File: src/components/common/TmBalance.vue [hunk 4]
```diff
@@ -41,14 +41,15 @@
...
 import refetchNetworkOnly from "scripts/refetch-network-only"
+import { shortDecimals } from "scripts/num"
 import { noBlanks } from "src/filters"
-import TmBtn from "common/TmBtn"
-import SendModal from "src/ActionModal/components/SendModal"
-import ModalWithdrawRewards from "src/ActionModal/components/ModalWithdrawRewards"
 import { UserTransactionAdded } from "src/gql"
 import { mapGetters } from "vuex"
+const TmBtn = () => import("common/TmBtn")
+const SendModal = () => import("src/ActionModal/components/SendModal")
```
Comment (faboweb) [inline src/components/common/TmBalance.vue]: let's load modals async and make sure they are not rendered directly (v-if) but let's remove the noise for the other components?

File: src/routes.js (hunk 3 above)
Comment (jbibla) [reply src/routes.js]: i don't think so. why do you think that?

**Comment by jbibla** [comment]
i'm going to close this in favour of a much cleaner PR #3207 that accomplishes the same goal with only a couple changes. i would propose that in the future someone investigates the following:

- reducing size of cosmos-ledger.js and cosmos-keys.js
- v-if -ing the action modals (which is not trivial at first glance)
- more vue / webpack optimization options

**Pull request closed without merging**